apis/v1beta1/types.go

```diff
@@ -50,12 +50,14 @@ type IdentityType string
 // Supported identity types.
 const (
 	IdentityTypeGoogleApplicationCredentials = "GoogleApplicationCredentials"
+
+	IdentityTypeAzurePrincipalCredentials = "AzurePrincipalCredentials"
```

Comment:
Principal is a generic term for referring to an authentication subject. I named it like this in case we support other principals like MSI or workload identity and not directly suggest Service Principals. I am OK with changing to `AzureServicePrincipalCredentials` or some other name